Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -52,48 +53,273 @@ public static class CustomEntitySearch
+                                    }
+                                    wordFound.GetStringBuilder().Clear();
+                                    currMismatch = 0;
+                                }
+                            }
+                        }
+                        else
+                        {
+                            string accents = @"[\p{Mn}\p{Sk}\p{Mc}]";
+                            bool wordHalfMark = Regex.Match(wordCharArray[currWordChar].ToString(), accents).Success;
+                            bool textHalfMark = Regex.Match(textCharArray[currTextChar].ToString(), accents).Success;
+
+                            // fuzzy situation?
+                            // accent case adds 0.5
+                            if ((wordHalfMark && textHalfMark) ||
+                                String.Compare(wordCharArray[currWordChar].ToString(), textCharArray[currTextChar].ToString(),
+                                CultureInfo.CurrentCulture, CompareOptions.IgnoreCase | CompareOptions.IgnoreNonSpace) == 0)
+                            {
+                                currMismatch += 0.5;
+                                wordFound.Write(text.ElementAt(currTextChar));
+                                currTextChar++;
+                                currWordChar++;
+                            }
+                            else if (currTextChar < textCharArray.Count - 1 && String.Compare(wordCharArray[currWordChar].ToString(), 
+                                textCharArray[currTextChar+1].ToString(), CultureInfo.CurrentCulture, CompareOptions.IgnoreCase | CompareOptions.IgnoreNonSpace) == 0)
+                            {
+                                currMismatch += (textHalfMark) ? 0.5 : 1;
+                                wordFound.Write(text.ElementAt(currTextChar));
+                                currTextChar++;
+                            }
+                            else if (currWordChar < wordCharArray.Count - 1 && String.Compare(wordCharArray[currWordChar + 1].ToString(),
+                                textCharArray[currTextChar].ToString(), CultureInfo.CurrentCulture, CompareOptions.IgnoreCase | CompareOptions.IgnoreNonSpace) == 0)
+                            {
+                                currMismatch += (wordHalfMark) ? 0.5 : 1;
+                                currWordChar++;
+                            }
+                            else
+                            {
+                                currMismatch += 1;
+                                wordFound.Write(text.ElementAt(currTextChar));
+                                currTextChar++;
+                                currWordChar++;
                             }
 
+                            if((currWordChar >= wordCharArray.Count || (currTextChar >= textCharArray.Count
+                                && currMismatch + (wordCharArray.Count - currWordChar) <= leniency))  && currMismatch <= leniency)
```

Comment:
>&& currMismatch <= leniency [](start = 102, length = 27)

isn't thi covered by the expression to the left of it?

Reply:
yes sorry!